x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardIndexer.java

```diff
@@ -545,12 +551,13 @@ public XContentBuilder buildDownsampleDocument() throws IOException {
             }
             builder.field(timestampField.name(), timestampFormat.format(timestamp));
             builder.field(DocCountFieldMapper.NAME, docCount);
+            // TODO: should we make sure extracting dimension fields is backward compatible with older indices versions?
             // Extract dimension values from _tsid field, so we avoid loading them from doc_values
-            Map<?, ?> dimensions = (Map<?, ?>) DocValueFormat.TIME_SERIES_ID.format(tsid);
```

Comment:
I think this logic (and TODO) can be removed? Because the dimensions can be extracted from the dimension doc values fields?